### PR TITLE
Replace typedefs benchmarks examples

### PR DESCRIPTION
**Pull request by masterleinad**
Part of #3213.

### CODE REVIEW

File: example/tutorial/04_simple_memoryspaces/simple_memoryspaces.cpp [hunk 1]
```diff
@@ -61,12 +61,12 @@ typedef Kokkos::View<double * [3]> view_type;
 // performance penalties then it is its own host_mirror_space. This is
 // the case for HostSpace, CudaUVMSpace and CudaHostPinnedSpace.
 
-typedef view_type::HostMirror host_view_type;
+using host_view_type = view_type::HostMirror;
```
Comment (ndellingwood) [inline example/tutorial/04_simple_memoryspaces/simple_memoryspaces.cpp:64]: I'm kind of surprised this doesn't require `typename`?

File: example/tutorial/05_simple_atomics/simple_atomics.cpp [hunk 2]
```diff
@@ -48,17 +48,17 @@
 #include <cmath>
 
 // Type of a one-dimensional length-N array of int.
-typedef Kokkos::View<int*> view_type;
-typedef view_type::HostMirror host_view_type;
+using view_type      = Kokkos::View<int*>;
+using host_view_type = view_type::HostMirror;
```
Comment (ndellingwood) [inline example/tutorial/05_simple_atomics/simple_atomics.cpp:52]: I'm kind of surprised this doesn't require `typename`?

File: example/tutorial/05_simple_atomics/simple_atomics.cpp [hunk 3]
```diff
@@ -48,17 +48,17 @@
...
 // value (an int, in this case).  Access the value using operator()
 // with no arguments: e.g., 'count()'.
 //
 // Zero-dimensional Views are useful for reduction results that stay
 // resident in device memory, as well as for irregularly updated
 // shared state.  We use it for the latter in this example.
-typedef Kokkos::View<int> count_type;
-typedef count_type::HostMirror host_count_type;
+using count_type      = Kokkos::View<int>;
+using host_count_type = count_type::HostMirror;
```
Comment (ndellingwood) [inline example/tutorial/05_simple_atomics/simple_atomics.cpp:61]: I'm kind of surprised this doesn't require `typename`?

File: example/tutorial/05_simple_atomics/simple_atomics.cpp [hunk 4]
```diff
@@ -118,7 +118,7 @@ int main() {
     host_view_type h_result = Kokkos::create_mirror_view(result);
     host_count_type h_count = Kokkos::create_mirror_view(count);
 
-    typedef view_type::size_type size_type;
+    using size_type = view_type::size_type;
```
Comment (ndellingwood) [inline example/tutorial/05_simple_atomics/simple_atomics.cpp:121]: I'm kind of surprised this doesn't require `typename`?

File: example/tutorial/Advanced_Views/02_memory_traits/memory_traits.cpp [hunk 5]
```diff
@@ -71,10 +71,10 @@ typedef Kokkos::View<double*> view_type;
...
 // same data you should not use them together if that brakes the const guarantee
 // of the first pointer.
-typedef Kokkos::View<const double*, Kokkos::MemoryTraits<Kokkos::RandomAccess> >
-    view_type_rnd;
-typedef Kokkos::View<int**> idx_type;
-typedef idx_type::HostMirror idx_type_host;
+using view_type_rnd =
+    Kokkos::View<const double*, Kokkos::MemoryTraits<Kokkos::RandomAccess> >;
+using idx_type      = Kokkos::View<int**>;
+using idx_type_host = idx_type::HostMirror;
```
Comment (ndellingwood) [inline example/tutorial/Advanced_Views/02_memory_traits/memory_traits.cpp:77]: I'm kind of surprised this doesn't require `typename`?

File: example/tutorial/Hierarchical_Parallelism/01_thread_teams/thread_teams.cpp [hunk 6]
```diff
@@ -57,12 +57,12 @@
 // league_size) is not limited by physical constraints. Its a pure logical
 // number.
 
-typedef Kokkos::TeamPolicy<> team_policy;
-typedef team_policy::member_type team_member;
+using team_policy = Kokkos::TeamPolicy<>;
+using team_member = team_policy::member_type;
```
Comment (ndellingwood) [inline example/tutorial/Hierarchical_Parallelism/01_thread_teams/thread_teams.cpp:61]: I'm kind of surprised this doesn't require `typename`?

File: example/tutorial/Hierarchical_Parallelism/04_team_scan/team_scan.cpp [hunk 7]
```diff
@@ -48,11 +48,11 @@
...
 
-typedef Kokkos::DefaultExecutionSpace Device;
-typedef Kokkos::HostSpace::execution_space Host;
+using Device = Kokkos::DefaultExecutionSpace;
+using Host   = Kokkos::HostSpace::execution_space;
 
-typedef Kokkos::TeamPolicy<Device> team_policy;
-typedef team_policy::member_type team_member;
+using team_policy = Kokkos::TeamPolicy<Device>;
+using team_member = team_policy::member_type;
```
Comment (ndellingwood) [inline example/tutorial/Hierarchical_Parallelism/04_team_scan/team_scan.cpp:55]: I'm kind of surprised this doesn't require `typename`?

**Review by ndellingwood — APPROVED** [review]
lgtm, there some places in various `example/tutorial/` files that I marked where I would expect a `typename` necessary to prevent compiler issues, but those were there beforehand, this PR just replaced `typedef` usage with `using`

**Review by jrmadsen — APPROVED** [review]
Looks good to me. Glad to see this done!

File: example/tutorial/04_simple_memoryspaces/simple_memoryspaces.cpp (hunk 1 above)
Comment (dalg24) [reply example/tutorial/04_simple_memoryspaces/simple_memoryspaces.cpp:64]: @masterleinad did you check whether that file gets compiled at all?

Comment (masterleinad) [reply example/tutorial/04_simple_memoryspaces/simple_memoryspaces.cpp:64]: Yes, that file gets compiled and the compiler doesn't complain with the new version.

File: example/tutorial/05_simple_atomics/simple_atomics.cpp (hunk 3 above)
Comment (masterleinad) [reply example/tutorial/05_simple_atomics/simple_atomics.cpp:61]: Same here.

File: example/tutorial/Hierarchical_Parallelism/04_team_scan/team_scan.cpp (hunk 7 above)
Comment (dalg24) [reply example/tutorial/Hierarchical_Parallelism/04_team_scan/team_scan.cpp:55]: because it is not a template

**Pull request merged**